### DISCUSSION TITLE
ET4: Pi shutdown stops pump but keeps filter circuit on (and Chlorinator running? 😬)

### DISCUSSION THREAD

**Discussion by johnny2678** [tagyoureit/nodejs-poolController#760] [App and O/S Config Questions]
Wanted to bring up this observation as I've been told that a chlorinator that's on while the pump isn't running is a bad thing?

I had just shut down the pi to do my [Sequent Smartfan retrofit](https://github.com/tagyoureit/nodejs-poolController/discussions/759) and when I went out to the pad, the pump wasn't running but the filter relay still showed on and the swg lights were still all on.

I quickly threw it into service mode and that killed the SWG power until i had the pi stack running again.

Happy to try and reproduce but wanted to ask what you would need to see.  Or if this is even an issue at all.

Would a replay during the shutdown help?  not sure that would capture the moment the pump goes off... real time video?

**Comment by rstrouse** [comment]
With an EasyTouch 4 njsPC is not in charge of the filter relay.  The OCP controls that.  Are you sure you have it wired to the correct side of the relay?

**Reply by johnny2678** [reply]
Haven't touched the OCP wiring since it was installed 8 years ago. Pump comes on changes speed and turns off on the schedule set by the ET4 so I assume it's wired correctly.

For some reason, the pump shuts down when the raspberry pi does 🤷‍♂️

Takes a full pi shutdown tho... pump will still run if pi is up and njspc restarts or shuts down.

**Reply by tagyoureit** [reply]
> the pump wasn't running but the filter relay still showed on and the swg lights were still all on.

The SWG has the built-in flow switch.  Were the lights showing chlorine generation or no flow?

**Reply by rstrouse** [reply]
This is definitely not anything njsPC is in control of.  When lets have a replay so I can look at your config.

**Reply by johnny2678** [reply]
[ok, took a vid so you could see what's happening.](https://youtu.be/t50gRT1GMbs)

Sorry for the bad videography, and my 8am voice after a long mothers day and baby duty last night.  I didn't do any editing so you could see how it all played out over 2.5 minutes.

Timestamps as follows:
00:08 - Replay Started
------- Pre- Pi Shutdown --------
00:22 - Confirmed Pump Running
00:25 - Confirmed OCP is controlling pump
00:31 - Confirmed Filter Pump Circuit is active on OCP
00:41 - Confirmed SWG running w/ power

00:51 - Raspberry Pi Shutdown cmd issued
-------- waiting -------------------
01:51 - After 1 full minute (interesting), the pump shuts down
01:57 - You can see pump is now under local control, not under OCP control
02:12 - Filter Pump Circuit still showing on
02:22 - SWG still has power, flow switch showing no flow tho
-------- NOT SHOWN IN VIDEO -------------
- cycled through service mode and timeout mode back to auto mode
- pump started again, this time back under OCP controll
- powered the Pi back up

2 observations that may not make this a big deal:
1. I'm not sure how dangerous this is, since if the Pi loses power, the pump/OCP also will not have power
2. the SWG flow switch is registering no flow, so hopefully that keeps the SWG from dosing anything while the pump is off, even though the SWG still has power

Sorry for the fire drill if this is nothing, just wanted to bring it up in case there was something in the code causing the pump to shut down even when an OCP is present.

I did grab a replay, but not sure it would grab the relevant events since the pump power off happens after the pi is shutdown.

[replay (2).zip](https://github.com/tagyoureit/nodejs-poolController/files/11478699/replay.2.zip)

**Reply by rstrouse** [reply]
Interesting.  There is no place in your entire replay where njsPC sends any commands to the pump.  It is entirely reliant on EasyTouch for that communication.  The 1 minute timeframe is about right for the OCP to stop communicating and release control of the pump.  The OCP wasn't in service mode by chance was it.  If it was then this is absolutely normal behavior.  Service mode releases the pump but retains control of the relays.  If you had turned off that relay would it cut power to the chlorinator?

**Reply by johnny2678** [reply]
No, the OCP was is auto mode during this whole video.  I only cycled to service mode when the video stopped to ensure that a) power to the SWG was cut, and b) cycle back around to auto mode to start the pump again.

Not surprised there is nothing in the replay because it happens one minute after the pi shuts down.

> If you had turned off that relay would it cut power to the chlorinator?

Sorry, do you mean turn off the filter pump circuit/relay?  If so, I haven't tried that but I can.  I usually have at least one rpm circuits running so would have to make sure those are off too.

**Reply by johnny2678** [reply]
> The 1 minute timeframe is about right for the OCP to stop communicating and release control of the pump.

1 minute after... what?  Still trying to figure out how the OCP is aware of the Pi shutting down.

**Reply by rstrouse** [reply]
At no point did njsPC send any messages to the pump.  It is not in control of it in any way.  Only the OCP (ET 4) communicates with your pump in your current configuration so there is no reason that the communication should stop if you shut down the pi.   I have verified that the only message to the pump are listed below and the entire conversation is between the pump and OCP.

The last command before the shutdown was to set the speed to 2310rpm.
![image](https://github.com/tagyoureit/nodejs-poolController/assets/47839015/27e79bb1-e3a2-48e6-ba0e-70535e954435)


What is characteristic is that if the pump stops getting those commands it will eventually release itself and return to manual control which is what is happening.

**Reply by johnny2678** [reply]
Agree that this isn't njspc.  

Just can't imagine what would keep the OCP from squawking after the Pi shuts down, when it fires right back up when I cycle back to auto mode and continues to work as expected without the pi present.

wonder if anyone else with an ET4/8 can test this.  Is it Ghosts in THE machine or just Ghosts in my machine? 🤔😂

**Reply by rstrouse** [reply]
I just tried it with my IntelliCenter and didn’t have any issues.

**Reply by rstrouse** [reply]
If at any time the OCP is in service mode it will stop squawking at the pump.

**Reply by johnny2678** [reply]
That makes sense, but when the pi shuts down, the OCP isn't in service mode. 

I only put it in service mode manually after the shutdown. 🤷‍♂️

**Reply by rstrouse** [reply]
Just making sure that wasn't the case.  If service mode isn't on then does the chlorinator control panel become enabled as well?  Just wondering if perhaps the comms go noisy with the pi shut down.  Which RS485 adapter do you have?  I know I have probably asked you this in the past but.... squirrel!

**Reply by johnny2678** [reply]
Sorry for the delay - was on the road this week.

>  If service mode isn't on then does the chlorinator control panel become enabled as well? Just wondering if perhaps the comms go noisy with the pi shut down. 

hmm... I don't have a SWG control panel.  SWG is controlled by ScreenLogic (if I wasn't using njsPC)

> Which RS485 adapter do you have? I know I have probably asked you this in the past but.... squirrel!

it's the Sequent Mega-BAS in passthrough mode

**Reply by rstrouse** [reply]
> hmm... I don't have a SWG control panel. SWG is controlled by ScreenLogic (if I wasn't using njsPC)

I am talking about the panel located on the cell.  Are the buttons disabled?

**Reply by johnny2678** [reply]
> I am talking about the panel located on the cell. Are the buttons disabled?

Will check but the family and I are headed out for some R&R for a week.  Will pick this back up when we get back!

**Comment by mhmatttu** [comment]
I experienced similar behavior (pi shutdown turned off pump/circuits) in the past and found it was an issue with MQTT retained messages.  If you have any of those interfaces enabled, you might check into that.
